File: 15_Assignment/Question2.c
#include<stdio.h>
#include<stdlib.h>

int FirstOcc(int Arr[],int iLength,int iNo){
    int iCnt=0;

    for(iCnt=0;iCnt<iLength;iCnt++){
        if((Arr[iCnt])==iNo){
            return iCnt;
        }
    }

    return -1;
}


int main(){
    int iSize=0;
    int *p=NULL;
    int iCnt=0;
    int iRet=0;
    int iValue=0;

    printf("Enter the elements that you want to entered :\n");
    scanf("%d",&iSize);

    p=(int*)malloc(iSize*sizeof(int));
    if(p==NULL){
        printf("Unable to allocate memory");
        return -1;
    }

    printf("Dynamic memory is allocated......\n");
    printf("Enter the %d elements :\n",iSize);

    for(iCnt=0;iCnt<iSize;iCnt++){
        printf("Enter the element %d\n",iCnt+1);
        scanf("%d",&p[iCnt]);
    }

    printf("Enter number that you want to search :\n");
    scanf("%d",&iValue);

    iRet=FirstOcc(p,iSize,iValue);

    if(iRet==-1){
        printf("There is no such number");
    }
    else{
        printf("First occurrence of number is %d",iRet);
    }

    

    free(p);




    return 0;
}

File: 35_Assignment/Question2.java
import java.util.*;

class Number {
    public int Brr[];
    public int iNo;

    public Number(int Brr[], int iNo) {
        this.Brr = Brr;
        this.iNo = iNo;
    }

    public int FirstOcc() {
        int iCnt = 0;
        int iPos = 1;

        for (iCnt = 0; iCnt < Brr.length; iCnt++) {
            if (Brr[iCnt] == iNo) {
                break;
            }
            iPos++;
        }
        return iPos;
    }
}

class Question2 {
    public static void main(String args[]) {
        Scanner sobj = new Scanner(System.in);

        int iSize = 0;
        int iCnt = 0;
        int iCheck = 0;
        int iRet = 0;
        System.out.println("Enter the number of elements : ");
        iSize = sobj.nextInt();

        System.out.println("Enter the number that you want to check: ");
        iCheck = sobj.nextInt();

        int Arr[] = new int[iSize];
        System.out.println("Enter the elements : ");
        for (iCnt = 0; iCnt < iSize; iCnt++) {
            Arr[iCnt] = sobj.nextInt();
        }

        Number ADobj = new Number(Arr, iCheck);
        iRet = ADobj.FirstOcc();
        System.out.println("First Occuercence : " + iRet);

        sobj.close();

    }
}